DuckPAN Test: default to prove, add --build for dzil test

moollaza opened the pull request:
This ensures that any users who run duckpan test won't run into dzil dependency issues (e.g. handlebars, etc)

Partially addresses #254 

/cc @zachthompson

zachthompson commented on lib/App/DuckPAN/Cmd/Test.pm:
```diff
@@ -6,16 +6,24 @@ with qw( App::DuckPAN::Cmd );
+    if ($self->full) {
+        $self->app->emit_error("Could not find dist.ini.") unless -e 'dist.ini';
+        $ret = system('dzil test');
+        $self->app->emit_error('Could not begin testing. Is Dist::Zilla installed?') if $ret == -1;
```
We should just replace $ret == -1 with line 24.  Any non-zero exit value is an error.

On line 27 as well we should output an error if system fails.

moollaza replied:
> We should just replace $ret == -1 with line 24. Any non-zero exit value is an error.

@zachthompson I think we still want to capture the return value though so we can `return` it below?

moollaza replied:
> On line 27 as well we should output an error if system fails.

:+1:

moollaza commented:
@zachthompson I added in the error message when prove fails and simplified the check for a non-zero exist status.

Let me know if you think anything should be changed!

zachthompson commented on lib/App/DuckPAN/Cmd/Test.pm:
```diff
@@ -6,20 +6,29 @@ with qw( App::DuckPAN::Cmd );
+	if ($self->full) {
+		$self->app->emit_error("Could not find dist.ini.") unless -e "dist.ini";
+		$ret = system("dzil test");
+		$self->app->emit_error("Could not begin testing. Is Dist::Zilla installed?") if $ret;
```
I was just saying that there's no reason to catch the value and then test it.:
``` Perl
$self->app->emit_error("Could not begin testing. Is Dist::Zilla installed?") if $ret = system("dzil test");
```
Same with below.  Not a big deal, just more succinct.

moollaza replied:
:+1:

moollaza commented:
@zachthompson this should be good now!
